gcovr/tests/cmake_oos/Makefile

```diff
@@ -13,6 +13,9 @@ else
   CMAKE_GENERATOR := "MSYS Makefiles"
 endif
 
+all: export CC := $(notdir $(subst ',,$(CC)))
+all: export CXX := $(notdir $(subst ',,$(CXX)))
+all: export GCOV := $(notdir $(subst ',,$(GCOV)))
```

Comment:
Cmake has problems if a there is a full path with quotes on Windows, see https://github.com/gcovr/gcovr/runs/4633457207?check_suite_focus=true#step:12:427. The quotes are added by the 'shutil.join'. This line removes the quotes and the path.

Reply:
Ah, it seems the single quotes are added to escape backslashes in a Windows path. I suspect there might be a better way, though I'm not sure if `$(shell echo $(CC))` is really any better.

Using the `$(notdir ...)` function is probably incorrect in the context of #492. An `/absolute/path/to/gcc` might refer to a different compiler than `gcc`.

Transforming GCOV like this is unnecessary because it's not used by CMake (hopefully), and possibly incorrect in case of a Clang-based test.

I think the most reasonable way to deal with this is to NOT shell-escape the contents of the CC and CXX variables, and to assume that they contain the path to a compiler, or the name of a compiler. So I don't think there should be any change in the Makefiles. Instead, we should reconsider the changes in the Noxfile. Using absolute paths is fine, but this doesn't require adding the complexities of shell escaping. Instead, we might ensure that the values in these variables don't need any shell escaping. On Windows, this should be achievable with a `str.replace(os.path.sep, '/')`.

Reply:
The problem were the backslashes and the single quotes. I've removed the whole escaping and changed the place where the `gcov` argument was added in the clang toolchain.